src/app/core.py

```diff
@@ -199,48 +200,60 @@ def license_compare_helper(request):
         result['context'] = context_dict
         return result
 
+
 def ntia_check_helper(request):
     """
-       A helper function to check the NTIA elements in a given file in various formats.
+    A helper function to check the NTIA minimum elements in a given file.
     """
     ajaxdict = dict()
     context_dict = {}
     result = {}
     try:
         if request.FILES["file"]:
             # Saving file to the media directory
-            myfile = request.FILES['file']
+            myfile = request.FILES["file"]
             folder = f"{request.user}/{int(time())}"
             folder_path = os.path.join(settings.MEDIA_ROOT, folder)
-            folder_url = urljoin(settings.MEDIA_URL, folder + '/')
+            folder_url = urljoin(settings.MEDIA_URL, folder + "/")
             fs = FileSystemStorage(location=folder_path, base_url=folder_url)
             filename = fs.save(utils.removeSpecialCharacters(myfile.name), myfile)
             local_path = fs.path(filename)
             # Get other request parameters
             sbom_spec = "spdx2"
-            format_ = request.POST.get("format", "")
+            format_ = request.POST.get("format", "").strip()
             if format_.startswith("SPDX3"):
                 sbom_spec = "spdx3"
             compliance = request.POST.get("compliance", "ntia")  # Default: "ntia"
+
             # Call the Python SBOM Checker
             schecker = SbomChecker(
                 f"{str(local_path)}",
                 compliance=compliance,
                 sbom_spec=sbom_spec,
             )
+            # Capture the printed output
             old_stdout = sys.stdout
             temp_stdout = StringIO()
             sys.stdout = temp_stdout
             schecker.print_components_missing_info()
-            schecker.print_table_output()
             sys.stdout = old_stdout
-            retval = temp_stdout.getvalue().replace(",",", ").replace("\n","<br/>")
+
+            html = schecker.output_html()
+            schecker_printout = re.sub(r"\n+", "\n", temp_stdout.getvalue().strip())
+            retval = schecker_printout.replace(",", ", ").replace("\n", "<br/>")
```

Comment:
This looks good to me. I feel bad that there needs to be convoluted regular expressions after `schecker.output_html()`. But, like you alluded in the PR, perhaps this could be cleaned up in a future version?

Reply:
That line is because we still need information about missing components and it is not available yet in HTML output. So we still have to rely on the output of print_components_missing_info in stdout, and thus the regex treatment.

Yes. I will looking at how to put the missing component info into HTML output in the next version of ntia-conformance-checker.

Reply:
Open the issue in ntia-conformance-checker as https://github.com/spdx/ntia-conformance-checker/issues/320

Reply:
Thank you too for the review :) 
I will put this as resolved for now and we will follow up in ntia-conformance-checker repo.